scrapy_zyte_api/handler.py

```diff
@@ -81,30 +84,16 @@ async def _download_request(self, request: Request, spider: Spider) -> Response:
             )
             raise IgnoreRequest()
         self._stats.inc_value("scrapy-zyte-api/request_count")
-        headers = self._prepare_headers(api_response.get("httpResponseHeaders"))
         # browserHtml and httpResponseBody are not allowed at the same time,
         # but at least one of them should be present
         if api_response.get("browserHtml"):
             # Using TextResponse because browserHtml always returns a browser-rendered page
             # even when requesting files (like images)
-            return TextResponse(
-                url=api_response["url"],
-                status=200,
-                body=api_response["browserHtml"].encode(self._encoding),
-                encoding=self._encoding,
-                request=request,
-                flags=["zyte-api"],
-                headers=headers,
+            return ZyteAPITextResponse.from_api_response(
+                api_response, request=request, encoding=self._encoding
```

Comment:
I wonder what is self._encoding for - do you think we can drop it, instead of passing it around?

Reply:
Updated in https://github.com/scrapy-plugins/scrapy-zyte-api/pull/10/commits/d0dc08ddd06b6f21800ccd2dd93af199792c88e6